src/Microsoft.Graph/Requests/Helpers/ChunkedUploadProvider.cs

```diff
@@ -147,33 +149,54 @@ public async Task<DriveItem> UploadAsync(int maxTries = 3, IEnumerable<Option> o
                 if (uploadTries < maxTries)
```

Comment:
This retry logic will result in 3^3 retries (by default) for 429, 503 and 504 HTTP responses when the service doesn't return a successful result. The underlying HTTP client already has a RetryHandler that handles retries for 429, 503 and 504 responses.